holoviews/core/data/multipath.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+
+from ..util import max_range
+from .interface import Interface
+
+class MultiInterface(Interface):
+    """
+    MultiInterface allows wrapping around a list of tabular datasets
+    including dataframes, the columnar dictionary format or 2D tabular
+    NumPy arrays. Using the split method the list of tabular data can
+    be split into individual datasets.
+    """
+
+    types = ()
+
+    datatype = 'multi'
+
+    subtypes = ['dataframe', 'dictionary', 'array', 'dask']
+
+    @classmethod
+    def init(cls, eltype, data, kdims, vdims):
+        new_data = []
+        dims = {'kdims': eltype.kdims, 'vdims': eltype.vdims}
+        extra_kws = {}
+        for d in data:
+            d, interface, dims, extra_kws = Interface.initialize(eltype, d, kdims, vdims,
+                                                                 datatype=cls.subtypes)
```

Comment:
A couple of comments:

1. ``for d in data`` could iterate over some giant iterable (e.g an image array) if the wrong type of data is passed in. Shouldn't we do a little bit of type checking here? My basic assumption is that the input is a list.
2. Out of ``d, interface, dims, extra_kws``, the ``interface`` value is never used so that one could definitely be replaced with ``_``. This brings me to ....
3.  Both ``dims`` and ``extra_kws`` are being overwritten in this loop which means the final return of ``init`` corresponds to the values from the last interface initialized - if ``data`` isn't empty. Otherwise it  uses the ``dims`` you declare before the loop and ``{}`` for ``extra_kws``. I assume the interfaces can change in the loop but perhaps there is some sort of validation that can be done here to ensure consistency? Making the return of ``MultiInterface.init`` correspond to the return of the last interface used may work but strikes me as a little odd...